File: ios/BankWalletApp/NativeModules/Locale/LocaleModule.m

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LocaleModule, NSObject)
RCT_EXTERN_METHOD(getSystemLocale:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
+ (BOOL)requiresMainQueueSetup;
@end

File: ios/BankWalletApp/NativeModules/Locale/LocaleModule.swift

import Foundation
import React

@objc(LocaleModule)
class LocaleModule: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc func getSystemLocale(_ resolve: RCTPromiseResolveBlock,
                             rejecter reject: RCTPromiseRejectBlock) {
    let tag = Locale.preferredLanguages.first ?? "en-US"
    resolve(tag)
  }
}
